File: csharp/leap/Year.cs
using System;

class Year
{
    static void Main()
    {

    }
    /// <summary>
    /// is the passed year a leap year
    /// </summary>
    /// <param name="args"></param>
    /// <returns></returns>
    static Year()
    {
        
    }
    public static bool IsLeap(int year)
    {

        //if the year is divisible by 400 it is
        //return as such
        if (year % 400 == 0)
        {
            return true;
        }
        //if the year is divisible by 100, it is not a leap year
        if (year % 100 == 0)
        {
            return false;
        }
        //if we get here, check if the year is divisible by 4
        return year % 4 == 0;
    }
}

File: python/leap/Year.py
#checks if a year passed is a leap year
#all years diisible by 400 are leap years
#then all years divisible by 4 but not by 100 are also leap years


def is_leap_year(year):
    return year % 400 == 0 or (year % 4 == 0 and year % 100 != 0)